sympy/tensor/array/tests/test_array_comprehension.py

```diff
@@ -0,0 +1,14 @@
+from sympy.tensor.array.array_comprehension import ArrayComprehension
+from sympy.abc import i, j
+
+
+def test_array_comprehension():
+    a = ArrayComprehension(i*j, (i, 1, 3), (j, 2, 4))
+    b = ArrayComprehension(i, (i, 1, j))
+    c = ArrayComprehension(i*j, (i, 1, 3))
+    assert a.doit() == [[2, 3, 4], [4, 6, 8], [6, 9, 12]]
+    assert isinstance(b.doit(), ArrayComprehension)
+    assert b.subs(j, 3) == [1, 2, 3]
```

Comment:
why is `.subs` triggering the evaluation? I suppose it should create `ArrayComprehension(i, (i, 1, 3))`.

Reply:
I was thinking that `.subs ` try do expand the array (calling a doit() inside) if all variable has a specific bound or value. It should be `[3, 6, 9]` instead, I made a mistake.